### DISCUSSION TITLE
Arctic drone ops

### DISCUSSION THREAD

**Discussion by harrylerman** [dannyjacobs/ECHO#49] [General]
I started looking into Arctic operations for our MARS trip this summer. I found a few references:

These guys used a Pixhawk Black with a Here+. They call it a Here+ RTK but don't mention the base station at all and their picture is just the antenna, their writeup reads like they used it without the RTK capability. They had some issues, but they think it was mostly temperature-based for the IMU and that using the dual-GPS capability of the FC would help. Looks like the temperature problem they identified can be fixed by setting a mavlink parameter (easy to do). Notably, our Pixhawk Orange is supposed to be an improvement over the Pixhawk Black and one improvement is the IMU. 
https://www.diva-portal.org/smash/get/diva2:1449945/FULLTEXT01.pdf

These people used a Pixhawk (does not say what kind) in the Arctic (79th parallel off of Greenland) and didn't have any real problems with the flight controller. Their main issue seems to be they did their compass calibration on a ship (metallic environment) so they had some initial heading problems on takeoff but once they manually flew away from the ship it auto-corrected. 
https://www.mdpi.com/2073-4433/11/4/416

And these guys used expensive DJI drones at Eureka: their non-RTK drone had some altitude drift and they witnessed the drone start flying in circles several times. Their RTK-capable drone had no problems, even with the RTK functionality intentionally disabled. 
https://amt.copernicus.org/articles/14/7123/2021/amt-14-7123-2021.pdf

So it will be interesting to see how our Cube Orange performs at MARS both with and without RTK running. I am hopeful that with RTK enabled it will be ok, and it might be fine even without RTK given the Greenland peoples' observations, and given that the group that had trouble with the Pixhawk at Arctic latitudes were using an older Cube Black.

**Comment by dannyjacobs** [comment]
The primary issue reported by Brenner and Hultmar (your first reference) is orientation because of the temp dependent response of the mems gyro in the IMU. The IMU is usually backstopped by the magnetometer, but at the pole the planar part of the field is a lot smaller. What is the mavlink parameter that helps here? Addition of RTK will not improve this situation because that only improves position and not orientation. 

 B&H describe dual GNSS as an ideal solution but "hard". One reason is that Arducopter only supported (in 2020) the Trimble dual antenna board. Its almost two years later and we're baselining px4 not Arducopter. This is not unlike the situation for RTK a few years ago where the firmware hadn't yet caught up with the available sensors. 

Have you looked around for dual antenna solutions? Maybe the latest HERE supports dual antennas.

**Reply by harrylerman** [reply]
Just saw this response, sorry for the delay here. The parameter they suggest to alter is BRD_IMU_TARGTEMP, which I guess is the target temp of the IMU heater. The default is 45, and they concluded that with the cold arctic temperature it couldn't actually reach this temp, so the heater was just always on and causing fluctuations (apparently). Discussed on pgs 51 and 54 of Brenner and Hultmar. 

I don't think Here supports a dual antenna solution yet based on the following link, but there are some dual antenna RTK solutions out there for px4, it might be worth looking into them: https://docs.px4.io/master/en/gps_compass/rtk_gps.html

**Reply by harrylerman** [reply]
Then again there may be a simpler way: https://docs.px4.io/master/en/advanced_config/tuning_the_ecl_ekf.html#yaw-from-gps-velocity

**Comment by christopherbarbarie** [comment]
I found an [interesting blog post](https://deepsouthrobotics.com/2019/06/03/the-taming-of-the-u-blox-zed-f9p/) about using a base and rover setup using two ZED-F9P enabled GNSS boards that could be an option for determining heading in an Arctic drone operation. The most relevant part is section 3 where the author discusses how to implement the base and rover configuration.
